framework/base/src/storage/mappers/token_mapper_state.rs

```diff
@@ -0,0 +1,28 @@
+use crate::{
+    api::ManagedTypeApi,
+    codec::{
+        self,
+        derive::{NestedDecode, NestedEncode, TopDecode, TopEncode},
+    },
+    types::TokenIdentifier,
+};
+
+#[derive(NestedEncode, NestedDecode, TopEncode, TopDecode, Default)]
```

Comment:
This encoding is not backwards compatible with the old TokenIdentifier stored in the token mapper.

To preserve backwards compatibility, it is possible to define an encoding where "" means TokenMapperState::NotSet, "pending" or something for Pending, and the verbatim token identifier for TokenIdentifier